pkg/snapshot/overlay.go

```diff
@@ -299,8 +299,9 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 		return nil, err
 	}
 
-	// NOTE: If the image is in overlaybd format, the baselayer will be
-	// the metadata(in small size) and should not be fetched on-demand.
+	// If the layer is in overlaybd format, only its metadata needs to be downloaded and saved into snapshot dir.
+	// Return a "already exists" error to skip Apply and Commit, which is a default procedure of tar format.
```

Comment:
I think the most important step skipped is fetching, not only Apply and Commit. 
And I think adding a link to 'https://github.com/containerd/containerd/blob/master/docs/remote-snapshotter.md' is better than  a comment.